cmake_test/asserts/file_contains.cmake

```diff
@@ -0,0 +1,51 @@
+include_guard()
+include(cmake_test/asserts/file_exists)
+
+#[[[ Assert a file contains the specified text.
+#
+# Asserts that the file at the specified path contains the specified text.
+#
+# :param _afc_file: The file to check
+# :type _afc_file: Path
+# :param _afc_text: The text to check for
+# :type _afc_text: String
+#]]
+function(ct_assert_file_contains _afc_file _afc_text)
```

Comment:
Probably better to write a function `file_contains` that returns `true` if the file contains the text and `false` otherwise then write the asserts in term of that function. This makes it easier to maintain these asserts down the road.